### ISSUE TITLE
KEVM update blocked by now-failing proofs post function encoding update

### ISSUE THREAD

**Issue by ehildenb** [kframework/kore#2227]
[kore-exec-1.tar.gz](https://github.com/kframework/kore/files/5441931/kore-exec-1.tar.gz)

This file comes from a simple proof on https://github.com/kframework/evm-semantics/pull/912/commits/06c9c8b514b544057e27a81e568acd89eda58654.

The proof is from `tests/specs/functional/lemmas-no-smt-spec.k`, and looks like this:

```
    claim <k> runLemma ( 1 ==Int bool2Word( B:Bool ) )   => doneLemma ( B ==K true ) ... </k>
```

The error produced looks like this:

```
kore-exec: [2886042] Error (ErrorDecidePredicateUnknown):
    Failed to decide predicate:            
        /* Sfa */
        \not{R}(
            /* Spa */
            \equals{SortBool{}, R}(
                /* Fl Fn D Sfa */ VarB:SortBool{},
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortBool{}}("false")
            )
        )
    with side condition:
        /* D Sfa */ \top{_PREDICATE{}}()

```

**Comment by ttuegel** [comment]
@ehildenb Can you confirm that the `evm-semantics` build system still runs this spec without the SMT solver?

**Comment by ehildenb** [comment]
@ttuegel yes the build system passes `--haskell-backend-command 'kore-exec --smt=none'`, and if you look at `kore-exec.sh`, you can see the `--smt none` option. Anything else you'd like me to verify about that?

**Comment by ttuegel** [comment]
This was probably triggered by #2170 and would be fixed by #2174. I need to investigate that, but that's where I'll start.

**Comment by ana-pantilie** [comment]
What I've found so far is that during the application of [this](https://github.com/kframework/evm-semantics/blame/master/evm-types.md#L166) equation, we send the condition to the solver when we check the requires, [here](https://github.com/kframework/kore/blob/6ae3e32af15f688c899ad6ae0ff5442af7afe6a2/kore/src/Kore/Equation/Application.hs#L375), and this is what triggers the error because we're running this test without the solver.

**Comment by ttuegel** [comment]
I'm can't really explain how this ever worked. The cited rule _requires_ the SMT solver to be applied. My hope for fixing this right now is that, as part of #1986, the backend will apply `simplification` rules before _and_ after function definitions, so we can avoid that rule with the appropriate lemma.
